### ISSUE TITLE
review lab10 Enrico Capuano s317038

### ISSUE THREAD

**Issue by enricocapuano** [FarisanFekri/computational-intelligence#8]
First of all, the code is a good application of reinforcement learning principles.
You smartly used two different approaches, to explore different RL strategies in Tic Tac Toe game.

The first approach uses an action-value function and simplifies the process of the determination of a winner, transforming the game into a numerical puzzle.
This is very effective and clear.

In the second approach, you used in a very good way the Monte Carlo method with full game strategy. 
Although it increases the method complexity, it allows to go deeper in the solution space, finding more game states and outcomes.

Overall, the code shows a good understading of all RL principles, also using the magic square in an effective way.
Moreover, I really appreciate the representation of all results in a table, that is very clear to read and understand and allows the reader to compare different scores.

In conclusion, you did a very good job exploring different RL techniques, so well done!
I hope my review will be useful to you, thank you for reading me!
